packages/scalecube-services/tests/services/ServiceProxyMW.test.js

```diff
@@ -1,45 +1,107 @@
             .preRequest((req$) => {
-                return req$.map( req => {
-                    req.message.data.push("Idan");
-                    return req;
-                });
+                return req$
+                    .map((req) => {
+                        req.message.data.push("Idan");
+                        return req;
+                    });
             })
             .services(new GreetingService(), new GreetingService())
-            .build()
+            .build();
+
+        const greetingService = ms
             .proxy()
             .api(GreetingService)
             .create();
 
-
-        expect.assertions(1);
         return expect(greetingService.hello()).resolves.toEqual("Hello Idan");
     });
-    it("MW should get service definition and microservices", () => {
+
+    it("preRequest should get service definition and microservices", () => {
         expect.assertions(6);
 
         const ms = Microservices
             .builder()
-            .preRequest((message) => {
-                return message.map((msg=>{
-                    expect(msg.thisMs).toEqual(ms);
-                    expect(msg.meta).toEqual(GreetingService.meta);
-                    expect(msg.message.data).toEqual(["Idan"]);
-                    expect(msg.message.method).toEqual("hello");
-                    expect(msg.message.serviceName).toEqual("GreetingService");
-                    return msg;
-                }));
+            .preRequest((req$) => {
+                return req$
+                    .map((msg) => {
+                        expect(msg.thisMs).toEqual(ms);
+                        expect(msg.meta).toEqual(GreetingService.meta);
+                        expect(msg.message.data).toEqual(["Idan"]);
+                        expect(msg.message.method).toEqual("hello");
+                        expect(msg.message.serviceName).toEqual("GreetingService");
+                        return msg;
+                    });
             })
             .services(new GreetingService(), new GreetingService())
             .build();
-        const greetingService = ms.proxy()
+
+        const greetingService = ms
+            .proxy()
+            .api(GreetingService)
+            .create();
+
+        return expect(greetingService.hello("Idan")).resolves.toEqual("Hello Idan");
+    });
+
+    it("postResponse should change message", () => {
+        expect.assertions(7);
+
+        const ms = Microservices
+            .builder()
+            .postResponse((response, data) => {
+                expect(data.request.data).toEqual(["Idan"]);
+                data.request.data[0] += ", it\'s Igor";
+                expect(isObservable(response)).toBeTruthy();
+                expect(data.inst).toBeDefined();
+                expect(data.request.serviceName).toEqual("GreetingService");
+                expect(data.thisMs).toEqual(ms);
+                expect(data.meta).toEqual(GreetingService.meta);
+                return response;
```

Comment:
I would expect this will be the response

Reply:
@idanilt Sorry didn't get it. Can you please describe what do you mean? Thank you!

Reply:
@idanilt Sorry didn't get it. Can you please describe what do you mean? Thank you!